src/common/hooks/account/use-account-transactions-with-transfers.hooks.ts

```diff
@@ -0,0 +1,43 @@
+import { useQuery, UseQueryOptions } from 'react-query';
+import { useCurrentAccountStxAddressState } from '@store/accounts/account.hooks';
+import { useCurrentNetworkState } from '@store/network/networks.hooks';
+import { useAccountsApi } from '@store/common/api-clients.hooks';
+import { AddressTransactionWithTransfers } from '@stacks/stacks-blockchain-api-types';
+import { useAccountTransactionsWithTransfersState } from '@store/accounts/transactions-with-transfer.hooks';
+import { DEFAULT_LIST_LIMIT, QueryRefreshRates } from '@common/constants';
+import { PaginatedResults } from '@common/types';
+
+const QUERY_OPTIONS = {
+  refetchInterval: QueryRefreshRates.MEDIUM,
+  refetchOnMount: 'always',
+  refetchOnReconnect: 'always',
+  refetchOnWindowFocus: 'always',
+};
+
+export function useGetAccountTransactionsWithTransfer(reactQueryOptions: UseQueryOptions = {}) {
+  const principal = useCurrentAccountStxAddressState();
+  const { url: networkUrl } = useCurrentNetworkState();
+  const { accountsApi } = useAccountsApi();
+  const fetch = () => {
+    if (!principal) return;
+    return accountsApi.getAccountTransactionsWithTransfers({
+      principal,
+      limit: DEFAULT_LIST_LIMIT,
+    });
+  };
```

Comment:
With react query, don't you need to pull in the values from the fetcher from the `queryKey` of the fetcher?

Reply:
In my case there is no need,  I use onSuccess which passes the data directly
https://github.com/blockstack/stacks-wallet-web/blob/feat/show-stx-transfer-amount-from-contract/I1713/src/common/hooks/account/use-account-transactions-with-transfers.hooks.ts#L38
We could do it differently though.

Reply:
No it's okay:

```
 useQuery(['todos'], fetchAllTodos)
 useQuery(['todos', todoId], () => fetchTodoById(todoId))
 useQuery(['todos', todoId], async () => {
   const data = await fetchTodoById(todoId)
   return data
 })
 useQuery(['todos', todoId], ({ queryKey }) => fetchTodoById(queryKey[1]))
```

I was thinking we need to do the last way, but as long as `principal` is in the config it's okay